File: src/main/java/com/dochiri/hexagonal/presentation/auth/request/RegisterUserRequest.java
package com.dochiri.hexagonal.presentation.auth.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;

public record RegisterUserRequest(
        @NotNull
        @Email
        String email,

        @NotNull
        String password,

        @NotNull
        String name
) {
}

File: src/main/kotlin/com/dochiri/hexagonal/presentation/auth/request/RegisterUserRequest.kt
package com.dochiri.hexagonal.presentation.auth.request

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class RegisterUserRequest(
    @field:NotBlank
    @field:Email
    val email: String,

    @field:NotBlank
    @field:Size(min = 8, max = 100)
    val password: String,

    @field:NotBlank
    @field:Size(min = 2, max = 50)
    val name: String
)
